### ISSUE TITLE
urbanthesaurus.org

### ISSUE THREAD

**Issue by devth** [devth/yetibot#649]
A nice addition to urbandictionary. 

No API so use scraping. This could even be implemented in terms of a Yetibot alias. 

**UPDATE**: actually there is an API. This is how you could impl it terms of an alias:

```
!json http://urbanthesaurus.org/api/related?term=programmer | json path $..word
```
e.g.

```
!alias urbanthes = "json http://urbanthesaurus.org/api/related?term=$s | json path $..word"
```

(We'd need a urlencode command to fixup that `$s` on the end of it's more than one word)

**Comment by dawsonfi** [comment]
Hi @devth, i would like to help with this issue.

Would you kindly help me with some details on where can i start, in order to implement this?

**Comment by devth** [comment]
I haven't decided how we should include commands like this one that can be expressed completely as an alias.

1. We could ignore that fact and use a json parsing lib - more code to maintain
2. We could store aliases somewhere and have them loaded on startup - kind of cool, but need to figure out where to store them. e.g. should Yetibot have the ability to load aliases from a GitHub repo or a gist?

If 2 is too overwhelming you could just implement 1 for now. 1. should be pretty straightforward:

1. Use clj-http.client to to make a request with the search term - see [output coercion examples](https://github.com/dakrone/clj-http#output-coercion)
1. Pull the results out of the data structure and return the list to the user

**Comment by devth** [comment]
Fixed by #662
